rules/alerting.go

```diff
@@ -46,52 +46,23 @@ func (s AlertState) String() string {
 		return "pending"
 	case StateFiring:
 		return "firing"
-	default:
-		panic("undefined")
 	}
+	panic(fmt.Errorf("unknown alert state: %v", s))
 }
 
 const (
-	// StateInactive is the state of an alert that is either firing nor pending.
 	StateInactive AlertState = iota
-	// StatePending is the state of an alert that has been active for less than
-	// the configured threshold duration.
```

Comment:
Why did you remove all comments? It's not clear for a new reader to know what a pending alert is, that requires knowledge we can also simply comment here.